django_structlog/middlewares/request.py

```diff
@@ -146,14 +146,14 @@ def prepare(self, request):
             structlog.contextvars.bind_contextvars(correlation_id=correlation_id)
         ip, _ = get_client_ip(request)
         structlog.contextvars.bind_contextvars(ip=ip)
+        log_kwargs = {
+            "request": self.format_request(request),
+            "user_agent": request.META.get("HTTP_USER_AGENT"),
+        }
         signals.bind_extra_request_metadata.send(
-            sender=self.__class__, request=request, logger=logger
-        )
-        logger.info(
-            "request_started",
-            request=self.format_request(request),
-            user_agent=request.META.get("HTTP_USER_AGENT"),
+            sender=self.__class__, request=request, logger=logger, log_kwargs=log_kwargs
         )
+        logger.info("request_started", **log_kwargs)
```

Comment:
Could you add please the same logic to request_finished and request_failed events? So it would be possible to remove/modify their context (`code` and `request`) as well.

Reply:
there you go.

Reply:
Looks perfect. Many thanks!